### PR TITLE
Add inititializers to Int and UInt which take UnsafePointers and Unsa…

### PR DESCRIPTION
**Pull request by MichaelBuckley**
#### What's in this pull request?

This is a small addition of initializers for Int and Uint which convert from UnsafePointer and UnsafeMutablePointer.
#### Resolved bug number: ([SR-1115](https://bugs.swift.org/browse/SR-1115))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by MichaelBuckley** [comment]
It looks like there's a merge conflict. I'm working on it now.

File: stdlib/public/core/UnsafePointer.swift.gyb [hunk 1]
```diff
@@ -586,6 +586,22 @@ extension ${Self} {
     fatalError("unavailable function can't be called")
   }
 }
+
+extension Int {
+  @_transparent
+  @warn_unused_result
+  public init<U>(bitPattern: ${Self}<U>) {
```
Comment (gribozavr) [inline stdlib/public/core/UnsafePointer.swift.gyb]: @jrose-apple Given nullability of pointers landing, this initializer should accept an optional pointer, right?

File: stdlib/public/core/UnsafePointer.swift.gyb [hunk 2]
```diff
@@ -586,6 +586,22 @@ extension ${Self} {
     fatalError("unavailable function can't be called")
   }
 }
+
+extension Int {
+  @_transparent
```
Comment (gribozavr) [inline stdlib/public/core/UnsafePointer.swift.gyb]: I don't think you need `@_transparent` here -- please drop it unless it breaks something.

File: stdlib/public/core/UnsafePointer.swift.gyb (hunk 1 above)
Comment (MichaelBuckley) [inline stdlib/public/core/UnsafePointer.swift.gyb]: What would be the correct behavior of a null pointer? Setting the Int/UInt to 0, or making this a failable initializer and returning nil?

File: stdlib/public/core/UnsafePointer.swift.gyb (hunk 2 above)
Comment (MichaelBuckley) [inline stdlib/public/core/UnsafePointer.swift.gyb]: Will do, thanks.

File: stdlib/public/core/UnsafePointer.swift.gyb (hunk 1 above)
Comment (jrose-apple) [inline stdlib/public/core/UnsafePointer.swift.gyb]: 0, I think. That way it round trips with the {{init(bitPattern:)}} initializers on UnsafePointer.

**Comment by gribozavr** [comment]
There is an `unsafeBitCast` that you could replace with these new APIs in `_rawPointerToString` in `stdlib/public/core/Runtime.swift.gyb`.

**Comment by MichaelBuckley** [comment]
It appears the `unsafeBitCast` in `_rawPointerToString` takes a `RawPointer` instead of an `UnsafePointer`, but I will look for other `unsafeBitCast`s that can be converted.

**Comment by gribozavr** [comment]
You should be able to initialize an UnsafePointer from a RawPointer first.

**Comment by MichaelBuckley** [comment]
I have resolved the merge conflict and addressed the issues pointed out so far in the review.

**Comment by gribozavr** [comment]
@swift-ci Please test and merge

**Pull request merged**